utils/zerovec/src/varzerovec/owned.rs

```diff
@@ -115,6 +167,118 @@ impl<T: AsVarULE> VarZeroVecOwned<T> {
         &self.entire_slice
     }
 
+    // Invalidate and resize the data at an index, optionally inserting or removing the index.
+    // Also updates affected indices and the length.
+    // Returns a slice to the new element data - it doesn't contain uninitialized data but its value is indeterminate.
+    unsafe fn shift(&mut self, index: usize, new_size: u32, shift_type: ShiftType) -> &mut [u8] {
```

Comment:
praise: this is a _really good_ refactoring

Reply:
Thank you! It was a fun puzzle :)